torch/csrc/api/include/torch/nn/modules/embedding.h

```diff
@@ -39,15 +75,45 @@ class TORCH_API EmbeddingImpl : public torch::nn::Cloneable<EmbeddingImpl> {
   /// Changes to `EmbeddingOptions` *after construction* have no effect.
   EmbeddingOptions options;
 
-  /// The embedding table.
+  /// The embedding table
   Tensor weight;
 };
 
+class TORCH_API Embedding : public torch::nn::ModuleHolder<EmbeddingImpl> {
+public:
+    using torch::nn::ModuleHolder<EmbeddingImpl>::ModuleHolder;
+    static Embedding from_pretrained(torch::Tensor embeddings, c10::optional<EmbeddingOptions> options = c10::nullopt, bool freeze= true);
+};
+
+class TORCH_API EmbeddingBagImpl : public torch::nn::Cloneable<EmbeddingBagImpl> {
+  public:
+    EmbeddingBagImpl(int64_t num_embeddings, int64_t embedding_dim)
+      : EmbeddingBagImpl(EmbeddingBagOptions(num_embeddings, embedding_dim)) {}
+    explicit EmbeddingBagImpl(EmbeddingBagOptions options);
+
+    void reset() override;
+
+    /// Pretty prints the `EmbeddingBag` module into the given `stream`.
+    void pretty_print(std::ostream& stream) const override;
+
+    torch::Tensor forward(const Tensor& input, c10::optional<torch::Tensor> offsets = c10::nullopt,
+      c10::optional<torch::Tensor> per_sample_weights = c10::nullopt);
+
+    /// The `Options` used to configure this `EmbeddingBag` module.
+    EmbeddingBagOptions options;
+    /// The embedding table
+    Tensor weight;
+};
+
+class TORCH_API EmbeddingBag : public torch::nn::ModuleHolder<EmbeddingBagImpl> {
+public:
+    using torch::nn::ModuleHolder<EmbeddingBagImpl>::ModuleHolder;
+    static EmbeddingBag from_pretrained(torch::Tensor embeddings, c10::optional<EmbeddingBagOptions> options = c10::nullopt, bool freeze= true);
```

Comment:
```suggestion
    static EmbeddingBag from_pretrained(const torch::Tensor& embeddings, const c10::optional<EmbeddingBagOptions>& options = c10::nullopt, bool freeze = true);
```

Reply:
I made a change to pass embeddings variable by address and make it const, but I don't think similar change can be made for options because we are changing options later in the function at this line:
Embedding embedding = Embedding((*options)._weight(embeddings));

Reply:
Sounds great, we don't need to make options const then